cmd/root.go

```diff
@@ -0,0 +1,227 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	rootCmd = &cobra.Command{
+		Use:   "zombie-detector",
+		Short: "zombie-detector detects longly undeleted kubernetes resources",
+		RunE:  rootMain,
+	}
+)
+
+const (
+	errorFaildtoList      string = "the server could not find the requested resource"
+	errorMethodNotAllowed string = "the server does not allow this method on the requested resource"
+)
+
+var inClusterFlag bool
+var thresholdFlag string
+var pushgatewayEndpointFlag string
+
+func init() {
+	rootCmd.AddCommand(versionCmd)
+	rootCmd.Flags().BoolVar(&inClusterFlag, "incluster", true, "execute in cluster or not")
+	rootCmd.Flags().StringVar(&thresholdFlag, "threshold", "24h", "threshold of detection")
+	rootCmd.Flags().StringVar(&pushgatewayEndpointFlag, "pushgateway", "", "URL of Pushgateway's endpoint")
+
+}
+
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func setupClusterConfig(inCluster bool) (*rest.Config, error) {
```

Comment:
`config.GetConfig` is useful in this case. You may omit `incluster` option.

> If --kubeconfig is set, will use the kubeconfig file at that location.  Otherwise will assume running in cluster and use the cluster provided kubeconfig.
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/config#GetConfig